src/gun_tunnel.erl

```diff
@@ -449,6 +451,8 @@ ws_send(Frames, State=#tunnel_state{protocol=Proto, protocol_state=ProtoState},
 
 %% Internal.
 
+%% Returns an error on send errors, a state otherwise
+%% @todo Handle error command.
```

Comment:
We do?

Reply:
Ah right, merge conflict remains probably.

Reply:
In `gun_tunnel` there are still two of these:

```Erlang
			%% @todo Handle error result from Proto:init/4
			{ok, _, ProtoState} = Proto:init(ReplyTo, OriginSocket, OriginTransport,
				ProtoOpts#{stream_ref => StreamRef, tunnel_transport => tcp}),
```

Reply:
OK I will do those after merge. But the todo pointed to here can be removed.

Reply:
I've removed the todo.

Now I've also solved the last remaining send error by allowing `gun_tunnel:init/6` to return an error and handle it where it's called. This is in the last commit. If you have something else in mind, I can revert it.

Reply:
OK I think that one is only partial. We don't want the entire connection to go down just because there's an error in the tunnel. The tunnel might be using a different socket than the connection (TLS proxy). But with the current code it's probably OK for now. I will just add a todo noting this to be improved later.